File: leetcode/medium/55_Jump_Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int pos = nums[0];
        for(int i = 0 ; i < nums.size() ; i++){
            if(i > pos){
                return false;
            }
            int length = nums[i]+i;
            if(length > pos){
                pos = length;
            }
        }
        return true;
    }
};

File: leetcode/medium/55_Jump_Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxJump = 0
        i=0
        while i <= maxJump and i < len(nums):
            maxJump = max(i + nums[i],maxJump)
            i+=1
        if i >= len(nums):
            return True
        return False